src/consensus/params.cpp

```diff
@@ -36,32 +40,138 @@ namespace Consensus {
         }
     }
 
-     int Params::GetLastFoundersRewardBlockHeight(int nHeight) const {
+    int Params::HalvingHeight(int nHeight, int halvingIndex) const {
         // zip208
         // FoundersRewardLastBlockHeight := max({ height ⦂ N | Halving(height) < 1 })
+        //
         // Halving(h) is defined as floor(f(h)) where f is a strictly increasing rational
         // function, so it's sufficient to solve for f(height) = 1 in the rationals and
```

Comment:
These comments (both here and the explanatory equations below) need to be updated now that this function takes an arbitrary `halvingIndex` instead of `1`.